lib/service/local_notification_service.dart

```diff
@@ -0,0 +1,218 @@
+        lines,
+        contentTitle: "${activeNotifications.length - 1} Updates",
+        summaryText: "${activeNotifications.length - 1} Updates",
+      );
+      AndroidNotificationDetails groupNotificationDetails =
+          AndroidNotificationDetails(
+        channelGroup.id,
+        channelGroup.name,
+        channelDescription: channelGroup.description,
+        styleInformation: inboxStyleInformation,
+        setAsGroupSummary: true,
+        groupKey: channelGroup.id,
+        // onlyAlertOnce: true,
+      );
+      NotificationDetails groupNotificationDetailsPlatformSpefics =
+          NotificationDetails(android: groupNotificationDetails);
+      await _localNotificationService.show(
+          0, '', '', groupNotificationDetailsPlatformSpefics);
+    }
+  }
+
+  Future<List<PendingNotificationRequest>> pendingNotificationRequests() async {
+    return await _localNotificationService.pendingNotificationRequests();
+  }
+
+  Future<void> cancelNotificationById(String id) async {
+    await _localNotificationService.cancel(generateIntFromString(id));
+  }
+
+  Future<void> cancelAllNotifications() async {
+    await _localNotificationService.cancelAll();
+  }
+
+  void onDidReceiveLocalNotification(
+      int id, String? title, String? body, String? payload) async {
+    if (payload == null) {
+      return;
+    }
+    final message = message_class.Message.fromJson(
+        jsonDecode(utf8.decode(payload.runes.toList())));
+    onNotificationClick.add(message);
+  }
+
+  Future<String> handleAction(String actionModule, String actionTable) async {
+    final provider = providers[actionModule];
+    if (provider == null) {
+      return "";
+    }
+    final information = provider[actionTable];
+    if (information == null) {
+      return "";
+    }
+    final path = information.item1;
+    return path;
+  }
+
+  void onDidReceiveNotificationResponse(NotificationResponse response) async {
+    if (response.payload == null) {
+      return;
+    }
+    final message = message_class.Message.fromJson(
+        jsonDecode(utf8.decode(response.payload!.runes.toList())));
+    onNotificationClick.add(message);
+  }
+
+  void onNotificationClickListener(message_class.Message message) async {
+    if (message.actionModule != null && message.actionTable != null) {
+      final path =
+          await handleAction(message.actionModule!, message.actionTable!);
+      QR.to(
+          "fr.myecl.titan://$path?actionModule=${message.actionModule!}&actionTable=${message.actionTable!}");
+    }
+  }
+}
+
+@pragma("vm:entry-point")
+void onDidReceiveBackgroundNotificationResponse(
+    NotificationResponse response) async {
+  if (response.payload == null) {
+    return;
+  }
+  final message = message_class.Message.fromJson(
+      jsonDecode(utf8.decode(response.payload!.runes.toList())));
+  if (message.actionModule != null && message.actionTable != null) {
+    final provider = providers[message.actionModule];
```

Comment:
This code is similar to handleAction, why not use this function ?

Reply:
You are right !